### ISSUE TITLE
Status app is not responding when right click -> Quit the app after sleep

### ISSUE THREAD

**Issue by anastasiyaig** [status-im/status-desktop#2651]
**Steps:**

1. install latest master and create account
2. click yellow button Hide to collapse the app to dock
3. run any other app on foreground
4. lock the laptop and move it to sleep (i just close laptop)
5. wait for a while , usually several minutes
6. open the laptop and unlock
7. right click the status app in dock -> quit

**As a result**, the app can't be closed and becomes not responsive

https://user-images.githubusercontent.com/82375995/120779456-c1c54d80-c52f-11eb-9820-6cbc94f8851e.mov

**Comment by saledjenic** [comment]
@anastasiyaig does this happen every time? I followed your steps and wasn't able to reproduce the behaviour you described.

**Comment by anastasiyaig** [comment]
it happens to me all the time when i close the laptop and leave it for night having status app open. Then right click -> close and the app is irresponsive

**Comment by saledjenic** [comment]
Ok, will try few times more and wait longer, I didn't wait so long, usually up to 10 minutes, but was trying many times these days in different parts of the day. 

I'm thinking now even I manage to catch that bug, there is a qt class which provides an interface for receiving native events but even then at least part of the fix need to be handled in OS' native code. Some idea may be to handle for macOS system notification kIOMessageSystemWillSleep and do some checks before system go to sleep state, and handle kIOMessageSystemWillPowerOn notification to prepare the app and reset its data once system is back from the sleep state.  That way, I guess, we will have all messages, transactions correctly displayed and not miss any if it has happened while system was in a sleep state (the app was inactive).

But I am not sure is this completely possible in nim, maybe we should try, all in all this should be discussed before we start. 
@iurimatias any thoughts on this?

**Comment by iurimatias** [comment]
@saledjenic for me this happens, without the sleep part, right clicking -> quit is not working, it only works on the status icon in the top systray

**Comment by iurimatias** [comment]
closing as duplicate of #3112 same issue
